packages/react/src/components/UIShell/SideNav.js

```diff
@@ -45,33 +52,78 @@ const SideNav = React.forwardRef(function SideNav(props, ref) {
     'aria-labelledby': ariaLabelledBy,
   };
 
-  const assistiveText = expanded
-    ? t('carbon.sidenav.state.open')
-    : t('carbon.sidenav.state.closed');
+  // const assistiveText = expanded
+  //   ? t('carbon.sidenav.state.open')
+  //   : t('carbon.sidenav.state.closed');
 
   const className = cx({
     [`${prefix}--side-nav`]: true,
     [`${prefix}--side-nav--expanded`]: expanded,
     [`${prefix}--side-nav--collapsed`]: !expanded && isFixedNav,
     [customClassName]: !!customClassName,
     [`${prefix}--side-nav--ux`]: isChildOfHeader,
+    [`${prefix}--side-nav--hidden`]: !isPersistent,
   });
 
+  const headerNavClassName = cx({
+    [`${prefix}--side-nav__header-navigation`]: hasHeaderItems,
+    [`${prefix}--side-nav__header-navigation--hidden`]: !hasHeaderItems,
+  });
+
+  const [headerNavChildren, setHeaderNavChildren] = useState(null);
+
+  useEffect(() => {
+    hasHeaderItems
+      ? setHeaderNavChildren([...headerNavigationRef.current.children])
+      : null;
+  }, []);
+
   return (
     <nav
       ref={ref}
       className={`${prefix}--side-nav__navigation ${className}`}
       {...accessibilityLabel}
       onFocus={event => handleToggle(event, true)}
       onBlur={event => handleToggle(event, false)}>
-      {children}
-      {isFixedNav ? null : (
-        <SideNavFooter
-          assistiveText={assistiveText}
-          expanded={expanded}
-          onToggle={handleToggle}
-        />
-      )}
+      <SideNavItems>
+        <div className={headerNavClassName}>
+          {headerNavChildren !== null
+            ? headerNavChildren.map(i => {
+                if (
+                  i.classList.length === 0 ||
+                  i.classList[0] !== `${prefix}--header__submenu`
+                ) {
+                  const text = i.innerText;
+                  const url = i.children[0].attributes[0].nodeValue;
+                  return (
+                    <SideNavLink href={url} key={`${Math.random()}`}>
```

Comment:
Since these items won't be reordered, would it be better to add an `index` to the `map` function and use that instead of `Math.random()`? Still not ideal, but it seems like `Math.random()` can cause unnecessary re-renders and is being called each time the menu is opened/closed.

It seems like the best case scenario is if each `SideNavLink` was provided a unique id that we could use as the key. 

For reference: https://stackoverflow.com/questions/29808636/when-giving-unique-keys-to-components-is-it-okay-to-use-math-random-for-gener

Reply:
@jnm2377 A bit related here, any reason behind `<SideNav>` itself rendering children as opposed to letting user render children that we did before...?

Reply:
I think it is because they are coming directly from the header, as opposed to being created inside the `SideNav`, when the screen goes to mobile

Reply:
Yes, @aledavila and I both thought it was better to use the header components as a ref rather than make users write two instances of the same navigation.

Reply:
Also, @tw15egan can definitely update it to use the index